File: data/frama_c_problems/5.c
/*@
    requires x >= y && x > 0 && y > 0;
    requires \valid(r);
    ensures *r < y;
    ensures x == \result*y + *r;
    
*/
int fun(int x, int y , int *r) {
    *r = x;
    int d = 0;
    /*@
        loop invariant  *r == x - y*d;
        loop assigns *r, d;
        loop variant *r;
    */
    while (*r >= y) {
        *r = *r - y;
        d = d + 1;
    }
    return d;
}

File: RQ1_result/success_without_minimal_fix/5.rs
#[allow(unused_imports)]
use vstd::math::*;
use vstd::prelude::*;
use vstd::slice::*;
verus! {

#[verifier::loop_isolation(false)]
#[verifier::exec_allows_no_decreases_clause]
fn fun(x: i32, y: i32, r: &mut i32) -> (result: i32)
    requires
        x >= y && x > 0 && y > 0,
        true,
    ensures
        ((r)@) < y,
        x == result * y + ((r)@),
{
    *r = x;
    let mut d: i32 = 0;
    while *r >= y
        invariant
            ((r)@) == x - y * d,
        decreases ((r)@),
    {
        *r = *r - y;
        d = d + 1;
    }
    d
}

fn main() {
}

} // verus!
